domain/secretbackend/service/service_test.go

```diff
@@ -1005,6 +1006,86 @@ func (s *serviceSuite) TestWatchSecretBackendRotationChanges(c *gc.C) {
 	wC.AssertNoChange()
 }
 
+func (s *serviceSuite) TestGetModelSecretBackend(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	svc := newService(
+		s.mockState, s.logger, s.clock, nil,
+	)
+
+	modelUUID := modeltesting.GenModelUUID(c)
+	gomock.InOrder(
+		s.mockState.EXPECT().GetModelSecretBackend(gomock.Any(), modelUUID).Return("backend-name", coremodel.CAAS, nil),
+		s.mockState.EXPECT().GetModelSecretBackend(gomock.Any(), modelUUID).Return("backend-name", coremodel.IAAS, nil),
+		s.mockState.EXPECT().GetModelSecretBackend(gomock.Any(), modelUUID).Return("kubernetes", coremodel.CAAS, nil),
+		s.mockState.EXPECT().GetModelSecretBackend(gomock.Any(), modelUUID).Return("internal", coremodel.IAAS, nil),
+	)
+
+	backendID, err := svc.GetModelSecretBackend(context.Background(), modelUUID)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(backendID, gc.Equals, "backend-name")
+
+	backendID, err = svc.GetModelSecretBackend(context.Background(), modelUUID)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(backendID, gc.Equals, "backend-name")
+
+	backendID, err = svc.GetModelSecretBackend(context.Background(), modelUUID)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(backendID, gc.Equals, "auto")
+
+	backendID, err = svc.GetModelSecretBackend(context.Background(), modelUUID)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(backendID, gc.Equals, "auto")
+}
+
+func (s *serviceSuite) TestSetModelSecretBackend(c *gc.C) {
```

Comment:
No tests covering rest of contract or well known failure cases.

Reply:
There is no point to test those errors because they are not raised from the service itself.